airflow/providers/cncf/kubernetes/cli/kubernetes_command.py

```diff
@@ -41,10 +41,10 @@
 @providers_configuration_loaded
 def generate_pod_yaml(args):
     """Generate yaml files for each task in the DAG. Used for testing output of KubernetesExecutor."""
-    execution_date = args.execution_date
+    logical_date = args.logical_date
     dag = get_dag(subdir=args.subdir, dag_id=args.dag_id)
     yaml_output_path = args.output_path
-    dr = DagRun(dag.dag_id, execution_date=execution_date)
+    dr = DagRun(dag.dag_id, logical_date=logical_date)
```

Comment:
We need a try-except for 2.x. See code in databricks_workflow.py below.

Reply:
Added it

Reply:
Changed it.